markdown/3.x/en/documentation/single-hit-redirect.md

```diff
@@ -0,0 +1,109 @@
+Please keep in mind that this field stores boolean values. 
+
+  ```javascript
+ if (result.articleNumberSearch && result.records && result.records.length === 1) {
+    //redirect user
+}      
+```
+
+**NOTE** All other examples in this article use `resultArticleNumberStatus`. 
+
+
+### Preventing redirection after getting back from record page
+If `search-immediate` is enabled, users might find themselves in a situation where they are unable to go back to the search result page, as the single hit redirect will keep redirecting them to the record page.
+In order to prevent this some information has to be stored and passed between documents. This information is intended to inform the single hit redirect mechanism to stop redirecting users when they got back from the record page they were just redirected to.
+The easiest way to implement such a guard is to store a boolean flag after detecting an exact search.
+This flag should be set or unset, depending on whether it is in the storage.
+
+ ```javascript
+ if (localStorage.getItem('ff_redirect_on') !== '0') {
+    if (result.resultArticleNumberStatus === 'resultsFound' 
+        && result.records 
+        && result.records.length === 1) {
+            localStorage.setItem('ff_redirect_on', 0); 
+            // redirect user
+    }   
+} else {
+    localStorage.setItem('ff_redirect_on', 1); 
+}   
+ ```
+
+### Search from different locations
+The solution above works in general, but it has one flaw which may have to be fine-tuned, especially if your application framework is dedicated to creating non-SPA applications.
+Since the `ff_redirect_on` flag is already set to `0`, the next exact search will not be redirected.
+This might not be desired behaviour, because the next search might be called from the page the user is currently on.
+Setting only one flag will not cover this case.
+What could help is to set an additional value that stores which page the `ff_redirect_on` flag was set to `0`.
+It can be used to only prevent redirection if the current location is identical to the one stored in this value.
+The two locations being identical would mean that the user has returned to the previous page.
+  
+ ```javascript
+if (localStorage.getItem('ff_redirect_referer') !== window.location.href) {
+    localStorage.setItem('ff_redirect_referer', window.location.href);
+} else {                    
+    localStorage.setItem('ff_redirect_referer', undefined); 
+}   
+ ```
+
+**NOTE** The above example uses `window.location.href`. It is a framework agnostic solution and may not work in more complicated cases. Please check the routing scope of the framework or libraries you are using to best adapt the algorithm to detect the return to the redirecting page.
+
+Full example below.
+Please keep in mind that on specific pages you might need to use different conditions in order to cover all edge cases.
+That script should be placed in every document the `ff-searchbox` element is used in.
```

Comment:
Just to clarify: `That` refers to the script below, correct? If so, `This script` or `The following script` may be more suitable. _That_ feels like it's pointing somewhere different.
```suggestion
This script should be placed in every document the `ff-searchbox` element is used in.
```
```suggestion
The following script should be placed in every document the `ff-searchbox` element is used in.
```

Reply:
Yes, this sections tells about the script placed below